src/Portable.Xaml/Portable.Xaml.ComponentModel/DateTimeConverter.cs

```diff
@@ -45,8 +58,28 @@ public override object ConvertTo (ITypeDescriptorContext context, CultureInfo cu
 
                 return date.ToString(format, culture);
             }
-			return base.ConvertTo (context, culture, value, destinationType);
-		}
-	}
+            return base.ConvertTo(context, culture, value, destinationType);
+        }
+
+        private string RemoveUnnecessaryZeros(string dateFormatted, DateTimeKind kind)
+        {
+            string conciseDate = dateFormatted.Trim();
+            if (kind == DateTimeKind.Utc)
+            {
+                if (UtcTidyUpPattern.IsMatch(dateFormatted))
+                    conciseDate = UtcTidyUpPattern.Replace(dateFormatted, "Z");
+                if (conciseDate.EndsWith(".Z"))
+                    conciseDate = conciseDate.Replace(".Z", "Z");
```

Comment:
Maybe this should be `conciseDate = conciseDate.Remove(consiceDate.Length - 2, 1)` to be more efficient and to make sure it doesn't replace anything else in the string (though the odds for this one are very low)